File: java/socketd/src/main/java/org/noear/socketd/transport/core/HandshakeInternal.java
package org.noear.socketd.transport.core;

import java.util.Map;

/**
 * 握手信息内部接口
 *
 * @author noear
 * @since 2.0
 */
public interface HandshakeInternal extends Handshake {
    /**
     * 获取消息源
     */
    MessageInternal getSource();

    /**
     * 获取输出元信息
     */
    Map<String, String> getOutMetaMap();
}

File: swift/socketd/socketd/src/socketd/transport/core/HandshakeInternal.swift
//
//  HandshakeInternal.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

protocol HandshakeInternal : Handshake{
    /**
        * 获取消息源
        */
    func getSource() -> MessageInternal;
}
